modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkOperationModelParserSdk.java

```diff
@@ -292,134 +298,99 @@ private ComponentAst getBody() {
   }
 
   private Stream<OperationModel> expandOperationWithoutModel(Map<String, MuleSdkOperationModelParserSdk> operationModelParsersByName,
-                                                             Set<String> visitedOperations,
-                                                             ComponentAst componentAst) {
+                                                             Set<String> visitedOperations, ComponentAst componentAst,
+                                                             Predicate<ComponentAst> filterCondition,
+                                                             Predicate<ComponentAst> ignoreCondition) {
     final MuleSdkOperationModelParserSdk operationParser =
         operationModelParsersByName.get(componentAst.getIdentifier().getName());
 
     if (operationParser != null) {
-      return operationParser.getOperationModelsRecursiveStream(operationModelParsersByName, visitedOperations);
+      return operationParser.getOperationModelsRecursiveStream(operationModelParsersByName, visitedOperations, filterCondition,
+                                                               ignoreCondition);
     } else {
       return Stream.empty();
     }
   }
 
   private Stream<OperationModel> getOperationModelsRecursiveStream(Map<String, MuleSdkOperationModelParserSdk> operationModelParsersByName) {
+    return getOperationModelsRecursiveStream(operationModelParsersByName, componentAst -> false, componentAst -> false);
```

Comment:
When the filterExpression returns true, it means this component and everything inside it has to be filtered. That's why the default one returns false always, because it will never filter